File: ios/Sources/ACCore/Models/Elements/ListElement.swift
import Foundation
// MARK: - List

public struct ListElement: Codable, Equatable {
    public let type: String = "List"
    public var id: String?
    public var items: [CardElement]
    public var maxHeight: String?
    public var style: String?
    public var spacing: Spacing?
    public var separator: Bool?
    public var height: BlockElementHeight?
    public var isVisible: Bool?
    public var requires: [String: String]?
    
    public init(
        id: String? = nil,
        items: [CardElement],
        maxHeight: String? = nil,
        style: String? = nil,
        spacing: Spacing? = nil,
        separator: Bool? = nil,
        height: BlockElementHeight? = nil,
        isVisible: Bool? = nil,
        requires: [String: String]? = nil
    ) {
        self.id = id
        self.items = items
        self.maxHeight = maxHeight
        self.style = style
        self.spacing = spacing
        self.separator = separator
        self.height = height
        self.isVisible = isVisible
        self.requires = requires
    }
}

File: android/ac-core/src/main/kotlin/com/microsoft/adaptivecards/core/models/ListElement.kt
package com.microsoft.adaptivecards.core.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import kotlinx.serialization.Transient
import kotlinx.serialization.json.JsonElement

@Serializable
@SerialName("List")
data class ListElement(
    @Transient override val type: String = "List",
    override val id: String? = null,
    override val isVisible: Boolean = true,
    override val separator: Boolean = false,
    override val spacing: Spacing? = null,
    override val height: BlockElementHeight? = null,
    override val requires: Map<String, String>? = null,
    override val fallback: JsonElement? = null,
    val items: List<CardElement>,
    val maxHeight: String? = null,
    val style: String? = null
) : CardElement
